### PR TITLE
Don't lose mention type of canonical mentions

### PR DESCRIPTION
**Pull request by rgabbard**

### CODE REVIEW

File: src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt [hunk 1]
```diff
@@ -204,9 +202,14 @@ class ColdStart2AidaInterchangeConverter(
...
 
+            val justificationType = provenanceToMentionType[cs_assertion.justifications]
+                    .asSequence()
+                    .maxWith(Ordering.explicit(listOf(CANONICAL_MENTION, PRONOMIAL_MENTION,
+                            NOMINAL_MENTION, NORMALIZED_MENTION, NAME_MENTION)))!!
+
             registerJustifications(entityResource, cs_assertion.justifications,
                     typeAssertion, cs_assertion.string, confidence,
-                    justificationType = cs_assertion.mentionType.name)
+                    justificationType = justificationType.name)
```
Comment (berquist) [inline src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt]: Was the issue that the kind of mention in the `justificationType` was getting dropped?

Comment (rgabbard) [reply src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt]: Yes - in ColdStart the single "best" mention for an entity will have two entries - one with its actual mention type and one with `canonical_mention`.  The bug was that `canonical_mention` was clobbering the true mention types of the canonical mentions.

File: src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt [hunk 2]
```diff
@@ -204,9 +202,14 @@ class ColdStart2AidaInterchangeConverter(
 //                associate_with_system(realisNode)
 //            }
 
+            val justificationType = provenanceToMentionType[cs_assertion.justifications]
+                    .asSequence()
+                    .maxWith(Ordering.explicit(listOf(CANONICAL_MENTION, PRONOMIAL_MENTION,
```
[the remark this answers is not in the document]
Comment (rgabbard) [reply src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt]: Added a comment - clearer now?

[the remark this answers is not in the document]
Comment (berquist) [reply src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt]: Yes, I didn't know what `Ordering` did.

**Pull request merged**